src/VecSim/vec_sim_tiered_index.h

```diff
@@ -159,6 +168,80 @@ VecSimTieredIndex<DataType, DistType>::topKQuery(const void *queryBlob, size_t k
     }
 }
 
+template <typename DataType, typename DistType>
+
+VecSimIndexInfo VecSimTieredIndex<DataType, DistType>::info() const {
+    VecSimIndexInfo info;
+    VecSimIndexInfo backendInfo = this->backendIndex->info();
+    VecSimIndexInfo frontendInfo = this->frontendIndex->info();
+    info.algo = VecSimAlgo_TIERED;
+    info.commonInfo.indexLabelCount = this->indexLabelCount();
+    info.commonInfo.indexSize = this->indexSize();
+    info.commonInfo.memory = this->getAllocationSize();
+    info.commonInfo.isMulti = this->backendIndex->isMultiValue();
+    info.commonInfo.type = backendInfo.commonInfo.type;
+    info.commonInfo.metric = backendInfo.commonInfo.metric;
+    info.commonInfo.dim = backendInfo.commonInfo.dim;
+    info.commonInfo.blockSize = INVALID_INFO;
+    info.commonInfo.last_mode = backendInfo.commonInfo.last_mode;
+
+    info.tieredInfo.backendAlgo = backendInfo.algo;
+    switch (backendInfo.algo) {
+    case VecSimAlgo_HNSWLIB:
+        info.tieredInfo.backendInfo.hnswInfo = backendInfo.hnswInfo;
+        break;
+    case VecSimAlgo_BF:
+    case VecSimAlgo_TIERED:
+        assert(false && "Invalid backend algorithm");
+    }
+
+    info.tieredInfo.backendCommonInfo = backendInfo.commonInfo;
+    // For now, this is hard coded to FLAT
+    info.tieredInfo.frontendCommonInfo = frontendInfo.commonInfo;
+    info.tieredInfo.bfInfo = frontendInfo.bfInfo;
+
+    info.tieredInfo.backgroundIndexing = this->frontendIndex->indexSize() > 0;
```

Comment:
I think we should expose the number of zombies in the tiered index info section as well (num marked deleted)